subsys/net/lib/download_client/src/http_download_client.c

```diff
@@ -89,7 +89,32 @@ static int socket_sectag_set(int fd, int sec_tag)
 	return 0;
 }
 
-static int resolve_and_connect(const char *const host, int family, int sec_tag)
+static int socket_bind(int fd, const char *apn)
+{
+	int err;
+	size_t len;
+	struct ifreq ifr = {0};
+
+	__ASSERT_NO_MSG(apn);
+
+	len = strlen(apn);
+	if (len >= sizeof(ifr.ifr_name)) {
```

Comment:
This operation does not set the `errno`, and we log it in case of error. We'll log rubbish in this case.

Reply:
Good catch, I opted to move the printing of `errno` that was at the end of the function in the for loop that attempts to `connect()`. All other functions log the `errno` right after it happens, so logging it uncoditionally at the end was a bit redundant.